core/source/Number.mint

```diff
@@ -42,17 +42,21 @@ module Number {
   fun fromString (input : String) : Maybe(Number) {
     `
     (() => {
+      if (typeof #{input} !== 'string') {
+        return #{Maybe::Nothing}
+      }
+
```

Comment:
`input` here is guaranteed to be string, so that's redundant.

Reply:
@gdotdesign requested it [here](https://github.com/mint-lang/mint/pull/612#discussion_r1210182549) and [here](https://github.com/mint-lang/mint/pull/612#discussion_r1210226658). You even upvoted the second comment :-)

Reply:
I do not get why this shows "outdated" now. Did you remove this code? Please do not.

Reply:
I know it's redundant but I did request it, better to have it then someone creating an issue about it in the future.

Reply:
I'm lil' wary going down that round, since it would mean adding such guards all over the codebase - in a typed language after all.

@nilslindemann I can't find the comments you're referring to 😕

Reply:
I also was like, "why check for it, Mint will anyway take care of it" when I implemented it the first time. There are also performance considerations when these checks are done everywhere. I agree with Sija, in general the users should be on their own when they use backticks.

Reply:
@gdotdesign What's the reason for having such guard here?

Reply:
Just backticks :joy: I can let it go.

Reply:
Hahaha, so let it go plz, since if this passes I'm gonna question the type-safety of the language :P